### PR TITLE
added api and dal models

### PR DESCRIPTION
**Pull request by UrielYosef**

### CODE REVIEW

File: src/TisaBackend/TisaBackend.Domain/Enums/UserRank.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace TisaBackend.Domain.Enums
+{
+    public enum UserRank
```
Comment (lirazyehezkel) [inline src/TisaBackend/TisaBackend.Domain/Enums/UserRank.cs:3]: Rank is usually 1-5

File: src/TisaBackend/TisaBackend.Domain/Models/Airline.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+
+namespace TisaBackend.Domain.Models
+{
+    public class Airline
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        public IList<Airplane> Airplanes { get; set; }
```
Comment (lirazyehezkel) [inline src/TisaBackend/TisaBackend.Domain/Models/Airline.cs:10]: Why do we want to keep both the list of planes here and also to hold for each plane to which company it belongs?

File: src/TisaBackend/TisaBackend.Domain/Models/AirlineUserRank.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+namespace TisaBackend.Domain.Models
+{
+    public class AirlineUserRank
+    {
+        public UserRank Value { get; set; }
+        public string Review { get; set; }
+        public int UserId { get; set; }
+        public int AirlineId { get; set; }
+
+        public Airline Airline { get; set; }
```
Comment (lirazyehezkel) [inline src/TisaBackend/TisaBackend.Domain/Models/AirlineUserRank.cs:12]: Why do we need this?

File: src/TisaBackend/TisaBackend.Domain/Models/AirPlaneType.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿namespace TisaBackend.Domain.Models
+{
+    public class AirPlaneType
```
Comment (lirazyehezkel) [inline src/TisaBackend/TisaBackend.Domain/Models/AirPlaneType.cs:3]: AirPlaneType == planeModel from Word?

File: src/TisaBackend/TisaBackend.Domain/Models/Airplane.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿namespace TisaBackend.Domain.Models
+{
+    public class Airplane
+    {
+        public int Id { get; set; }
+        public int AirPlaneTypeId { get; set; }
+        public int AirlineId { get; set; }
+
+        public AirPlaneType AirPlaneType { get; set; }
+        public Airline Airline { get; set; }
```
Comment (lirazyehezkel) [inline src/TisaBackend/TisaBackend.Domain/Models/Airplane.cs:10]: Again why do we need this if we keep airlineId? same with type
I guess I just miss something

File: src/TisaBackend/TisaBackend.Domain/Models/AirPlaneType.cs (hunk 4 above)
Comment (UrielYosef) [reply src/TisaBackend/TisaBackend.Domain/Models/AirPlaneType.cs:3]: Yes (I didn't want to confuse with Models dir)

File: src/TisaBackend/TisaBackend.Domain/Models/Airline.cs (hunk 2 above)
Comment (UrielYosef) [reply src/TisaBackend/TisaBackend.Domain/Models/Airline.cs:10]: It is one to many and many to one relationship in efcore (represents the database)

File: src/TisaBackend/TisaBackend.Domain/Enums/UserRank.cs (hunk 1 above)
Comment (UrielYosef) [reply src/TisaBackend/TisaBackend.Domain/Enums/UserRank.cs:3]: Agree

File: src/TisaBackend/TisaBackend.Domain/Models/AirlineUserRank.cs (hunk 3 above)
Comment (UrielYosef) [reply src/TisaBackend/TisaBackend.Domain/Models/AirlineUserRank.cs:12]: need what?

**Pull request merged**